### PR TITLE
[INTERN-12] Cover app with DI

### PR DESCRIPTION
**Pull request by Strishwork**

### CODE REVIEW

File: app/src/main/java/com/example/countryinfo/CountryDetailsFragment.kt [hunk 1]
```diff
@@ -38,19 +40,14 @@ class CountryDetailsFragment : Fragment(), CountryDetailsAdapter.ViewHolder.OnIt
...
         super.onActivityCreated(savedInstanceState)
         val id = arguments?.get(COUNTRY_ID) as? String
-        val viewModel: CountryDetailsViewModel by viewModels {
-            CountryDetailsViewModelFactory(
-                CountriesApi(
-                    apolloClient(requireContext())
-                )
-            )
-        }
+        val viewModel: CountryDetailsViewModel by viewModels { viewModelFactory }
```
Comment (yeoliinyk) [inline app/src/main/java/com/example/countryinfo/CountryDetailsFragment.kt]: Let's move this to class level property.

File: app/src/main/java/com/example/countryinfo/CountryPreviewFragment.kt [hunk 2]
```diff
@@ -35,21 +36,13 @@ class CountryPreviewFragment : Fragment() {
...
-            this, CountriesPreviewViewModelFactory(
-                CountriesApi(
-                    apolloClient(
-                        requireContext()
-                    )
-                )
-            )
-        )
-            .get(CountriesPreviewViewModel::class.java)
+        val viewModel: CountriesPreviewViewModel by viewModels { viewModelFactory }
```
Comment (yeoliinyk) [inline app/src/main/java/com/example/countryinfo/CountryPreviewFragment.kt]: The same.

File: app/src/test/java/com/example/countryinfo/CountryDetailsViewModelTest.kt [hunk 3]
```diff
@@ -66,7 +66,7 @@ class CountryDetailsViewModelTest {
         assertThat(actualState.countryDetails.flag, `is`("https://test.com"))
         assertThat(actualState.countryDetails.currencyNames[0], `is`("Test currency"))
         assertThat(actualState.countryDetails.languages[0], `is`("Test language"))
-        assertThat(actualState.countryDetails.timezones[0], `is`("UTC+03:00"))
+        assertThat(actualState.countryDetails.timezones[0], `is`("UTC+05:00"))
```
Comment (yeoliinyk) [inline app/src/test/java/com/example/countryinfo/CountryDetailsViewModelTest.kt]: Do w need this change?

File: app/src/test/java/com/example/countryinfo/CountryDetailsViewModelTest.kt [hunk 4]
```diff
@@ -133,7 +133,7 @@ class CountryDetailsViewModelTest {
             `when`(flag?.svgFile).thenReturn("https://test.com")
             `when`(currencies).thenReturn(listOf(CountryDetails.Currency("", "Test currency", "")))
             `when`(officialLanguages).thenReturn(listOf(CountryDetails.OfficialLanguage("", "Test language")))
-            `when`(timezones).thenReturn(listOf(CountryDetails.Timezone("", "UTC+03:00")))
+            `when`(timezones).thenReturn(listOf(CountryDetails.Timezone("", "UTC+05:00")))
```
Comment (yeoliinyk) [inline app/src/test/java/com/example/countryinfo/CountryDetailsViewModelTest.kt]: Same here?

File: app/src/main/java/com/example/countryinfo/model/CountriesApi.kt [hunk 5]
```diff
@@ -8,8 +8,9 @@ import com.example.GetCountriesQuery
 import com.example.GetCountryByIdQuery
 import com.example.countryinfo.ICountriesApi
 import io.reactivex.Observable
+import javax.inject.Inject
 
-class CountriesApi(private val apolloClient: ApolloClient) :
+class CountriesApi @Inject constructor(private val apolloClient: ApolloClient) :
```
Comment (yeoliinyk) [inline app/src/main/java/com/example/countryinfo/model/CountriesApi.kt]:
```suggestion
class CountriesApi(private val apolloClient: ApolloClient) :
```

**Review by yeoliinyk — APPROVED** [review]
Looks good! Left minor nitpicks.

File: app/src/main/java/com/example/countryinfo/CountryDetailsFragment.kt (hunk 1 above)
Comment (Strishwork) [reply app/src/main/java/com/example/countryinfo/CountryDetailsFragment.kt]: Done.

File: app/src/main/java/com/example/countryinfo/CountryPreviewFragment.kt (hunk 2 above)
Comment (Strishwork) [reply app/src/main/java/com/example/countryinfo/CountryPreviewFragment.kt]: Done.

File: app/src/test/java/com/example/countryinfo/CountryDetailsViewModelTest.kt (hunk 3 above)
Comment (Strishwork) [reply app/src/test/java/com/example/countryinfo/CountryDetailsViewModelTest.kt]: Removed.

File: app/src/test/java/com/example/countryinfo/CountryDetailsViewModelTest.kt (hunk 4 above)
Comment (Strishwork) [reply app/src/test/java/com/example/countryinfo/CountryDetailsViewModelTest.kt]: Removed.

File: app/src/main/java/com/example/countryinfo/model/CountriesApi.kt (hunk 5 above)
Comment (Strishwork) [reply app/src/main/java/com/example/countryinfo/model/CountriesApi.kt]: Removed.

**Pull request merged**